balancer.go

```diff
@@ -89,27 +92,28 @@ type FairBalancer struct {
 	clusterstate ClusterState
 
 	releaseThreshold float64
-
-	lastreleased map[string]bool
+	delay            time.Duration
 }
 
 func (e *FairBalancer) Init(s BalancerContext) {
 	e.bc = s
 }
 
-// CanClaim will claim all tasks, but will add a sleep to block claiming
-// released tasks in order to give other nodes a chance to claim them first
-func (e *FairBalancer) CanClaim(taskid string) bool {
-	if e.lastreleased[taskid] {
-		time.Sleep(500 * time.Millisecond)
+// CanClaim will claim any task if this node wasn't over the task threshold
+// during the last rebalance. If it was over the threshold claims will be
+// delayed proportionally.
+func (e *FairBalancer) CanClaim(taskid string) (time.Time, bool) {
+	if e.delay == 0 {
+		return time.Time{}, true
```

Comment:
Nit alert!  could we make a const thats like `NoDelay`?   Just to make it read better?